test: add e2e tests for OpenAI server

Vasuk12 opened the pull request:
<!--
Thank you for your pull request. Please review and complete the sections below.
-->

### Which issue(s) does this pull-request address?

<!--
Please include a link to an issue in the tracker.  The issue describes the problem to be solved.  If there is no issue raised for this PR then either raise one with a summary and description of the problem or add a summary and description of the problem here
-->

Ref: #1217

### Description

<!-- Provide a description of the change, pay special attention to describing any breaking changes.  The description describes the resolution to the problem described in the linked issue (or to the problem outlined in this PR). -->

This PR adds end-to-end (e2e) tests for the OpenAI chat completion server API. The tests verify the complete request/response cycle between client and server, ensuring proper handling of successful requests, authentication failures, and model not found scenarios.

### Checklist

- Added `python/tests/e2e/test_openai_chat_completion.py` with three e2e test cases:
  - `test_openai_e2e_success`: Tests successful chat completion flow with valid API key
  - `test_openai_e2e_auth_failure`: Tests authentication failure with invalid API key (401 response)
 
### Future Imporvements

The following test cases will be added to enhance coverage:

1. **Missing API key test**: Testing behavior when no `Authorization` header is provided
2. **Streaming responses test**: Testing the streaming API endpoint with `stream: true`
3. **Invalid request format test**: Test handling of malformed JSON, missing required fields, and invalid message structures
4. **Response schema validation**: Add detailed validation to ensure response structure matches OpenAI API format (checking all required fields in the response)
5. **Concurrent requests test**: Test server behavior under concurrent load with multiple simultaneous requests

#### General
- [x] I have read the appropriate contributor guide: [Python](https://github.com/i-am-bee/beeai-framework/blob/main/CONTRIBUTING.md)
/ [TypeScript](https://github.com/i-am-bee/beeai-framework/blob/main/CONTRIBUTING.md)
- [x] I have signed off on my commit: [Python instructions](https://github.com/i-am-bee/beeai-framework/blob/main/python/CONTRIBUTING.md#developer-certificate-of-origin-dco) / [TypeScript instructions](https://github.com/i-am-bee/beeai-framework/blob/main/typescript/CONTRIBUTING.md#developer-certificate-of-origin-dco)
- [x] Commit messages and PR title follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary)
- [x] Appropriate label(s) added to PR: `Python` for Python changes, `TypeScript` for TypeScript changes

#### Code quality checks
- [x] Code quality checks pass: `mise check` (`mise fix` to auto-fix)

#### Testing
- [ ] Unit tests pass: `mise test:unit`
- [x] E2E tests pass: `mise test:e2e`
- [ ] Tests are included (for bug fixes or new features)

#### Documentation
- [ ] Documentation is updated
- [ ] Embedme embeds code examples in docs. To update after edits, run: Python `mise docs:fix`

xjacka commented on python/tests/e2e/test_openai_chat_completion.py:
```diff
@@ -0,0 +1,118 @@
+        api_key="TEST_KEY",
+    )
+
+    server = OpenAIServer(config=config, memory_manager=UnlimitedMemoryManager())
```
remove `memory_manager` for the completion endpoint

Vasuk12 commented on python/tests/e2e/__init__.py:
> Please move this to the `texts/adapters/openai` folder

I moved test_openai_chat_completion.py into the openai folder. Should I also proceed with deleting the entire e2e folder, including its __init__.py file?

Vasuk12 commented:
I've implemented the bug fix where the server now returns a 404 Not Found status code instead of a generic 500 Internal Server Error. I also added the 'test test_openai_e2e_model_not_found' to ensure this behavior is correctly validated.

Please review these changes. Let me know if there are any other related changes or edge cases you'd like me to address

xjacka commented on python/tests/adapters/openai/test_openai_chat_completion.py:
```diff
@@ -0,0 +1,127 @@
+
+@pytest.mark.e2e
+def test_openai_e2e_model_not_found(start_test_server: str) -> None:
+    url = f"{start_test_server}/chat/completions"
```
Can you add a responses API test too?

xjacka commented on python/beeai_framework/adapters/openai/serve/chat_completion/api.py:
```diff
@@ -68,7 +68,16 @@ async def handler(
-        runnable = self._model_factory(request.model)
+        try:
+            runnable = self._model_factory(request.model)
+        except RuntimeError as e:
```
You can change the error type here https://github.com/i-am-bee/beeai-framework/blob/main/python/beeai_framework/adapters/openai/serve/server.py#L80 . For example to: `raise HTTPException(status_code=404, detail=f"Model {model_id} not registered")`
